src/java.base/share/classes/jdk/internal/foreign/SegmentBulkOperations.java

```diff
@@ -309,8 +310,15 @@ private static long vectorizedMismatchLargeForBytes(MemorySessionImpl aSession,
 
     // The returned value is in the interval [0, 2^30]
     static int powerOfPropertyOr(String name, int defaultPower) {
-        final int power = Integer.getInteger(PROPERTY_PATH + name, defaultPower);
-        return 1 << Math.clamp(power, 0, Integer.SIZE - 2);
+        final String property = GetPropertyAction.privilegedGetProperty(PROPERTY_PATH + name);
```

Comment:
Hello Per, `sun.security.action.GetIntegerAction.privilegedGetProperty(PROPERTY_PATH + name, defaultPower)` could avoid the null checks and the try blocks here.

Reply:
Yes, the 2-arg GetIntegerAction.privilegedGetProperty would be better here, and it would retain the existing behavior for when the property value can't be parsed as a number.

Reply:
I think we need the try block anyhow as we have to deal with a String. But might be slightly better. We could revisit this later.

Reply:
It would look like this:

```
    static int powerOfPropertyOr(String name, int defaultPower) {
        final String property = GetPropertyAction.privilegedGetProperty(PROPERTY_PATH + name, Integer.toString(defaultPower));
        try {
            return 1 << Math.clamp(Integer.parseInt(property), 0, Integer.SIZE - 2);
        } catch (NumberFormatException _) {
            // ignore
        }
        return defaultPower;
    }
    ```

Reply:
https://bugs.openjdk.org/browse/JDK-8340089

Reply:
I think there's a oversight here. What we meant was a separate GetIntegerAction class instead of GetPropertyAction. The code would then look like:

```java
final int power = sun.security.action.GetIntegerAction.privilegedGetProperty(PROPERTY_PATH + name, defaultPower);
return 1 << Math.clamp(power, 0, Integer.SIZE - 2);
```

Reply:
https://github.com/openjdk/jdk/pull/20985

Thanks for spotting this.